eclair-core/src/test/scala/fr/acinq/eclair/TestConstants.scala

```diff
@@ -57,9 +62,62 @@ object TestConstants {
     }
   }
 
-  def sqliteInMemory() = DriverManager.getConnection("jdbc:sqlite::memory:")
+  sealed trait TestDatabases {
+    val connection: Connection
+    def network(): NetworkDb
+    def audit(): AuditDb
+    def channels(): ChannelsDb
+    def peers(): PeersDb
+    def payments(): PaymentsDb
+    def pendingRelay(): PendingRelayDb
+    def getVersion(statement: Statement, db_name: String, currentVersion: Int): Int
+    def close(): Unit
+  }
```

Comment:
Why don't you use the `Databases` trait? You can mix-in the `close` method if needed.

Reply:
1.  I needed a sealed trait for exhaustive pattern matching
2. Some of the tests need a fresh connection every time `network`, `audit`, etc gets called, but they are defined as `val` in `Databases`
3. `Databases` require `obtainExclusiveLock()` to be implemented, but it's database specific and can't be tested in a generic tests.